application/app/javascript/controllers/submit_button_controller.js

```diff
@@ -0,0 +1,26 @@
+import { Controller } from "@hotwired/stimulus"
+
+export default class extends Controller {
+    static targets = ["spinner", "label"]
+
+    click(event) {
+        event.preventDefault()
+
+        const uiDelay = window.loop_app_config.ui_feedback_delay
+
+        const button = event.currentTarget
+        button.disabled = true
+
+        const width = button.offsetWidth
+        const height = button.offsetHeight
+        button.style.width = `${width}px`
+        button.style.height = `${height}px`
```

Comment:
adding or removing CSS classes in the components is advised rather than injecting inline style properties. Can this feature be implemented with CSS classes instead?

Reply:
No - this needs to be like this. This is to avoid a jump in the front end when displaying the spinner.